### DISCUSSION TITLE
Design treatment and messaging for "Login with ATProto"

### DISCUSSION THREAD

**Discussion by bmann** [ATProtoApps/community#12] [General]
Starting from this [issue thread](https://github.com/ATProtoApps/community/issues/3), we had a live call today with a number of people sharing design concepts, as well as narrowing down what we want to achieve.

We talked about protocol logos and a bunch of other things, but ultimately scoped this down to "how do we share a base approach for logins that says to users that they can enter an ATProto handle and login".

The ideal outputs would be a symbolm, some help/info text, as well as default HTML/CSS/JS. Developers could drop this into their applications and have some visual and interaction similarity. A call to action to login with ATProto.

Further directions might include a protocol logo (but: git has a protocol logo no one knows, and use Github or Gitlab), some consideration for how OAuth scopes for specific apps look like, or other protocol level design and branding elements.

Please share any imagery in this thread, as well as links to mockups, code, or other elements.

**Comment by bmann** [comment]
![Screen Shot 2024-12-12 at 11 32 23 AM](https://github.com/user-attachments/assets/002cc133-3dae-4127-ae54-f16068ac373f)

Here's an example login for https://recipe.exchange

* just says handle with the @
* specifically says to go sign up with Bluesky and come back
* app specific colours, you have to know what this is

**Comment by ebwinters** [comment]
side note: whatever design chosen, it would be helpful to update the [statusphere example app](https://atproto.com/guides/applications) to use that common library or implement the design 

This is where a lot of devs start tinkering so having the right login components from the start will help people get going faster and using common frameworks. I'm personally having a little trouble getting the repo owner's attention since I've added jetstream ingestion

**Reply by bmann** [reply]
Great point @ebwinters

**Comment by bmann** [comment]
<img width="252" alt="Screen Shot 2024-12-16 at 7 32 14 PM" src="https://github.com/user-attachments/assets/b4ada3b7-9c03-4822-97ee-eab4810e7ad9" />

Here's an example from [WhiteWind](https://whtwnd.com/), the long form blogging app 
* sign in with Bluesky account -> leads to next screenshot
* sign up goes straight to Bluesky app (which, to be fair, lets you do third party sign up! so that's something else to consider...)

<img width="435" alt="Screen Shot 2024-12-16 at 7 33 49 PM" src="https://github.com/user-attachments/assets/6b22fbb5-28a3-4cc8-945f-6cc37907fde9" />

* labeled with "Bluesky handle or DID"
* `<username>.bsky.social` as placeholder
* it _does_ support self hosted PDS

**Comment by bnewbold** [comment]
Quick notes:

- we are scheduling work to enable account sign-ups directly on PDS/entryway, without going through the Bluesky app itself. On the Bluesky-hosted servers, this will still probably be branded "Bluesky" (as in the company) in some way, but for independent/self-hosted PDS instances it will be more generic. Devs will be welcome to point "Sign Up" links to the Bluesky PDS instances, and that flow will be a bit more app-generic; or of course they can direct to some other PDS hosting. This is a first step towards moving a bunch of account management (change email, change handle, delete account, etc) over to PDS interface. We don't plan to remove the possibility of doing those actions from apps (via APIs), but think they live better in PDS. Signup specifically can be complex (invite codes, captcha and other anti-abuse, SMS, 2FA enrollment, passkeys, password managers, etc) and may be hard to support in-app
- I don't want it to be "sign in with Bluesky", but at least mentioning Bluesky probably does reduce confusion and increase success.
- ActivityPub/Fediverse/Mastodon is a good analogy. I bet most casual less-techy folks don't know what "ActivityPub" is. If I was doing fediverse login i'd say something like "Login with a Fediverse account (eg, Mastodon)". So maybe "Login via Atmosphere (eg, Bluesky)"

**Comment by bmann** [comment]
From @verdverm

https://bsky.app/profile/verdverm.com/post/3lh55fhi2m22t

![IMG_8695](https://github.com/user-attachments/assets/546a3ac1-87c8-4f23-aab0-b596c51f72bf)


- Login with AT Proto
- Shows Bluesky
- default text with default bsky handle

**Comment by verdverm** [comment]
@bmann I've made a few adjustments to the above (ignoring that there are two inputs, and some spacing issues)

The main thing is to align with the ATProto logo (still just text & css)
- spell out the full ATProtocol
- `@AT` are all blue, the rest black

---

<img width="540" alt="Screenshot 2025-02-08 at 12 35 53 PM" src="https://github.com/user-attachments/assets/593cc576-80ee-46f1-8eb6-b20869d10506" />
